build-tools/src/main/java/org/elasticsearch/gradle/testclusters/TestClustersRegistry.java

```diff
@@ -22,11 +22,16 @@ public abstract class TestClustersRegistry implements BuildService<BuildServiceP
     private static final String TESTCLUSTERS_INSPECT_FAILURE = "testclusters.inspect.failure";
     private final Boolean allowClusterToSurvive = Boolean.valueOf(System.getProperty(TESTCLUSTERS_INSPECT_FAILURE, "false"));
     private final Map<ElasticsearchCluster, Integer> claimsInventory = new HashMap<>();
+
     private final Set<ElasticsearchCluster> runningClusters = new HashSet<>();
 
     public void claimCluster(ElasticsearchCluster cluster) {
         cluster.freeze();
-        claimsInventory.put(cluster, claimsInventory.getOrDefault(cluster, 0) + 1);
+        int claim = claimsInventory.getOrDefault(cluster, 0) + 1;
+        claimsInventory.put(cluster, claim);
+        if (claim > 1) {
+            cluster.setShared(true);
```

Comment:
We keep that information in the serialised cluster as the BuildServiceParameters cannot be keeping state across builds

Reply:
a Task itself is basically cached. So (nested) beans or fields of a task need to be cacheable

Reply:
Therefore keep the information in the cluster